### PR TITLE
[Sorting feature] First prototype of sorting feature

### PR DESCRIPTION
**Pull request by Rodolfodc**
This code is the initial proposal of the sorting feature.

This Code follows the suggestion of ordering by the following sequence:

1. data
2. variable
3. resource
4. misc

Please, let me know if you don't agree with this initial step, we are here to improve and make things work better, and don't forget that we will come back in this feature to implement new steps and improve it's usability.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -42,24 +44,127 @@ export function activate(context: vscode.ExtensionContext) {
...
+      position: number
+    };
+
+    let dataList: any[] = [];
+    let varList: any[] = [];
+    let resList: any[] = [];
+    
+    console.log(">>>>>>>>>>>> Datas")
+    console.log(typesMap.get("data")!.length);
+    console.log(typesMap.get("data"));
```
Comment (m-yassuo) [inline src/extension.ts:79]: Same as previous comment.

File: src/extension.ts [hunk 2]
```diff
@@ -42,24 +44,127 @@ export function activate(context: vscode.ExtensionContext) {
...
       if (reservedMainKeywords.includes(arrResult[i].line[0].value)) {
         arrResult[i].mainType = arrResult[i].line[0].value;
       } else {
         arrResult[i].mainType = "attribution";
       }
+
+      let currValue: Array<any> = [];
+
+      if(typesMap.has(arrResult[i].mainType)) {
+        console.log("Map content to be retrieved: ", typesMap.get(arrResult[i].mainType));
```
Comment (m-yassuo) [inline src/extension.ts:58]: I believe these lines may have been used for debug purposes, but are not needed.
For readability purposes, we could change this into comments.

File: src/extension.ts [hunk 3]
```diff
@@ -42,24 +44,127 @@ export function activate(context: vscode.ExtensionContext) {
...
+
+      dataList.sort((d1, d2) => {
+        return d1.element.line[2].value.toLowerCase().localeCompare(d2.element.line[2].value.toLowerCase());
+      });
+    }
+    console.log("datalist: ", dataList);
+
+    console.log(">>>>>>>>>>>> Variables");
+    console.log(typesMap.get("variable")!.length);
+    console.log(typesMap.get("variable"));
```
Comment (m-yassuo) [inline src/extension.ts:97]: Same as previous comment.

File: src/extension.ts [hunk 4]
```diff
@@ -42,24 +44,127 @@ export function activate(context: vscode.ExtensionContext) {
...
       vscode.window.showInformationMessage('found elements: ' + element.line.join(' '));
     });
 
     //rewrite file
+    let burnedIndex = new Array<number>();
     const fd = fs.openSync(editor.document.fileName, 'w');
     let filePos = 0;
     console.log('writing blocks');
-    writeBlocks(fd, filePos, arrResult);
+    console.log('writing data...');
```
Comment (m-yassuo) [inline src/extension.ts:137]: Same as previous comment.

**Review by pedro-carranza — APPROVED** [review]
LGTM!

File: src/orderer.ts [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            resultList = this.sortListByTFElement(
+                this.iterateElementArray(elementKey, arrOrigin),
+                elementSortIndex
+            );
+        }
+
+        return resultList;
+    }
+
+    protected sortListByTFElement(unsortedList: any[], index: number): any[] {
```
Comment (m-yassuo) [inline src/orderer.ts:58]:
```suggestion
    protected sortListByTfElement(unsortedList: any[], index: number): any[] {
```
This is just a question I have, which convention should we adopt?
Also a link for the discussion
https://stackoverflow.com/questions/15526107/acronyms-in-camelcase

**Review by m-yassuo — APPROVED** [review]

File: src/orderer.ts (hunk 5 above)
Comment (Rodolfodc) [reply src/orderer.ts:58]: Good point @m-yassuo I didn't notice it before! I will fix it!

**Pull request merged**